core/sync.js

```diff
@@ -296,6 +296,9 @@ class Sync {
       } else {
         await this.doMove(side, doc, from)
       }
+      if (!sameBinary(from, doc)) {
+        await side.overwriteFileAsync(doc, from)
+      }
```

Comment:
Well spotted!

What happens when calling `sameBinary()` for a moved directory is not obvious, although it actually works.
But the whole `#applyDoc()` and `#do*()` methods need refactoring (out of the scope of this PR).

We're calling `Side#overwriteFileAsync()` directly, which means we save an `if`, but also `transfer-started` won't be emitted... Could this be an issue?

Actually I'm also wondering why we're emitting `*-started` events after the local/remote writes are complete...
Could this be because some old code didn't use `async` / `await` hence the events emitted before writes completion?

Reply:
Looks like we just reused the event, but it has a new meaning of "transfer-made"

https://github.com/cozy-labs/cozy-desktop/commit/b3b3e446998d67744e64dfad765847f35b56bc15